documentation/src/docs/wire-protocol.md

```diff
@@ -7,106 +7,103 @@ You may obtain a copy of the License at
+# Reading
 
-### KeepAlive - Request/Reply
+## Read Segment - Request
 
-1.  Data - uninterpreted data of the length of the message. (Usually 0 bytes)
+1. `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8. This Segment indicates the Stream Segment that was read.
+2. `Offset`: Long (8 bytes). The offset in the Stream Segment to read from.
+3. `Reply`: Integer (4 bytes). The clients can request for the required length to the server (but the server may allot a different number of bytes).
+4. `delegationToken`: String (2 byte) followed by that many bytes of Java's Modified UTF-8. This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-Reading
--------
 
-### Read Segment - Request
+## Segment Read - Reply
 
-1.  Segment to read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset to read from - Long (8 bytes)
-3.  Suggested Length of Reply - int (4 bytes)
-    1.  This is how much data the client wants. They won't necessarily get that much.
+1. `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8). This Segment indicates the Stream Segment that was read.
+2. `Offset`: Long (8 bytes). The offset in the Stream Segment to read from.
+3. `Tail`: Boolean (1 bit). If the read was performed at the tail of the Stream Segment.
+4. `EndOfSegment`: Boolean (1 bit). If the read was performed at the end of the Stream Segment.
+5. `Data`: Binary (remaining length in the message).
 
-### Segment Read - Reply
+The client requests to read from a particular Segment at a particular Offset. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server may decide transferring to the client more or less data than it was asked for, splitting that data in a suitable number of reply messages.
 
-1.  Segment that was read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset that was read from - Long (8 bytes)
-3.  Is at Tail - Boolean (1 bit)
-4.  Is at EndOfSegment - (1 bit)
-5.  Data - Binary (remaining length in message)
+# Appending
 
-The client requests to read from a particular stream at a particular offset, it then receives one or more replies in the form of SegmentRead messages. These contain the data they requested (assuming it exists). The server may decided to give the client more or less data than it asked for, in as many replies as it sees fit.
+## Setup Append - Request
+1. `RequestId`: Long (8 bytes). This field contains the client-generated _ID_ that has been propagated to identify a client request.
+2. `writerId`: UUID (16 bytes). It identifies the requesting appender.
+3. `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8). This Segment indicates the Stream Segment that was read.
```

Comment:
Extra `)` in first sentence. Please, check all these extra parenthesis, as this typo seems to be repeated several times.

Reply:
@RaulGracia Edited in all occurences